### PR TITLE
Adds some git methods

### PR DESCRIPTION
**Pull request by fedefernandez**
This PR adds some needed `git` methods for adding new content to a GitHub repo.

Please @juanpedromoreno, could you review? Thanks

### CODE REVIEW

File: github4s/jvm/src/test/scala/github4s/unit/ApiSpec.scala [hunk 1]
```diff
@@ -289,4 +290,148 @@ class ApiSpec
...
+    val response =
+      gits.reference(accessToken, headerUserAgent, validRepoOwner, validRepoName, validRefSingle)
+    response should be('right)
+
+    response.toOption map { r =>
+      r.statusCode shouldBe okStatusCode
+    }
+  }
+
+  it should "return the multiple reference" in {
```
Comment (juanpedromoreno) [inline github4s/jvm/src/test/scala/github4s/unit/ApiSpec.scala]: "return multiple references"?

File: github4s/shared/src/main/scala/github4s/Decoders.scala [hunk 2]
```diff
@@ -157,4 +160,34 @@ object Decoders {
       )
   }
 
+  val emptyListDecodingFailure = DecodingFailure("Empty Response", Nil)
+
+  implicit def decodeListRef(implicit D: Decoder[Ref]): Decoder[NonEmptyList[Ref]] = {
+
+    def decodeCursor(partialList: Option[NonEmptyList[Ref]], cursor: HCursor): Decoder.Result[NonEmptyList[Ref]] =
+      cursor.as[Ref] match {
```
Comment (juanpedromoreno) [inline github4s/shared/src/main/scala/github4s/Decoders.scala]: `map`?

File: github4s/shared/src/main/scala/github4s/Decoders.scala [hunk 3]
```diff
@@ -157,4 +160,34 @@ object Decoders {
...
+
+    def decodeCursors(cursors: List[HCursor]): Result[NonEmptyList[Ref]] =
+      cursors.foldLeft[Decoder.Result[NonEmptyList[Ref]]](Left(emptyListDecodingFailure)) {
+        case (Right(list), cursor)                      => decodeCursor(Some(list), cursor)
+        case (Left(`emptyListDecodingFailure`), cursor) => decodeCursor(None, cursor)
+        case (Left(e), _)                               => Left(e)
+      }
+
+    Decoder.instance { c ⇒
+      c.as[Ref] match {
```
Comment (juanpedromoreno) [inline github4s/shared/src/main/scala/github4s/Decoders.scala]: `bimap`?

File: github4s/shared/src/main/scala/github4s/Decoders.scala (hunk 2 above)
Comment (fedefernandez) [inline github4s/shared/src/main/scala/github4s/Decoders.scala]: 👍

**Pull request merged**